test/logger/test_rotation.ts

```
@@ -0,0 +1,73 @@
+
+  it("deletes oldest files beyond maxFiles", async () => {
+    await enforceMaxFiles(dir, 2);
+    const remaining = await readdir(dir);
+    assert.ok(remaining.includes("server-2026-04-12.log"));
+    assert.ok(remaining.includes("server-2026-04-11.log"));
+    assert.ok(!remaining.includes("server-2026-04-10.log"));
+    assert.ok(remaining.includes("unrelated.txt")); // non-log files untouched
+  });
+
+  it("is a no-op when maxFiles is 0 or negative", async () => {
+    const before = await readdir(dir);
```

ESLint

[error] 'before' is already declared in the upper scope on line 1 column 17.

(no-shadow)